src/Shared/TrimmingAttributes.cs

```diff
@@ -154,6 +154,153 @@ public UnconditionalSuppressMessageAttribute(string category, string checkId)
     public string? Justification { get; set; }
 }
 
+/// <summary>
+/// States a dependency that one member has on another.
+/// </summary>
+/// <remarks>
+/// This can be used to inform tooling of a dependency that is otherwise not evident purely from
+/// metadata and IL, for example a member relied on via reflection.
+/// </remarks>
+[AttributeUsage(
+    AttributeTargets.Constructor | AttributeTargets.Field | AttributeTargets.Method,
+    AllowMultiple = true, Inherited = false)]
+internal sealed class DynamicDependencyAttribute : Attribute
```

Comment:
Why do we need this via shared source?

Reply:
These attributes were only introduced in `net5.0`. You can see what versions they are supported in https://apisof.net/catalog/5e049fae-038e-f176-2964-3db4b6b04ad5. DataProtection builds for `netstandard2.0` and `net4xx` where these attributes aren't defined. So when building for those older TFMs, we copy the attributes into the assembly so we don't need to use a bunch of `#if` defs.

Reply:
TIL :)